Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -296,24 +289,30 @@ private class BetterHTTPParser {
             // does not meet the requirement of RFC 7230. This is an outstanding http_parser issue:
             // https://github.com/nodejs/http-parser/issues/251. As a result, we check for these status
             // codes and override http_parser's handling as well.
-            if self.requestHeads.count > 0 {
-                let method = self.requestHeads.removeFirst().method
-                if method == .HEAD || method == .CONNECT {
-                    return .skipBody
+            if let method = self.requestHeads.popFirst()?.method {
```

Comment:
This whole thing seems like it would read better as a `guard let`, not an `if let`. That will also let us keep the `method` and `statusCode` if blocks separate as they were before, which I find is substantially less mentally taxing to understand.